src/timer.rs

```diff
@@ -0,0 +1,70 @@
+trait Timer: Sized {
+    fn start(milliseconds: u32) -> Self;
+    fn has_elapsed(self) -> Option<Self>;
```

Comment:
Let's add some documentation for those 2 functions.

Reply:
Done.